engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -392,31 +420,40 @@ public void endWindow()
 
   public void writeTuple(Object obj, String port)
   {
+    ++totalTupleCount;
     if (windowIdRanges.isEmpty()) {
       throw new RuntimeException("Data tuples received from tuple recorder before any BEGIN_WINDOW");
     }
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+    Slice f = null;
     try {
-      ByteArrayOutputStream bos = new ByteArrayOutputStream();
-      Slice f = streamCodec.toByteArray(obj);
-      PortInfo pi = portMap.get(port);
-      String str = "T:" + System.currentTimeMillis() + ":" + pi.id + ":" + f.length + ":";
-      bos.write(str.getBytes());
-      bos.write(f.buffer, f.offset, f.length);
-      bos.write("\n".getBytes());
-      PortCount pc = portCountMap.get(port);
-      pc.count++;
-      portCountMap.put(port, pc);
-
-      storage.writeDataItem(bos.toByteArray(), true);
-      //logger.debug("Writing tuple for port id {}", pi.id);
-      //fsOutput.hflush();
-      ++totalTupleCount;
-      if (numSubscribers > 0) {
-        // this is not asynchronous.  we need to fix this
-        publishTupleData(pi.id, obj);
+      f = streamCodec.toByteArray(obj);
+    } catch (RuntimeException ex) {
+      if (shouldLog(ex)) {
```

Comment:
can you log the exception inside `shouldLog` and return immediately. There is no need to check for `f != null` in this case.